src/optimizer/rules/implementation_rules.cpp

```diff
@@ -117,43 +113,85 @@ void LogicalGetToPhysicalIndexScan::Transform(common::ManagedPointer<AbstractOpt
-  if (sort != nullptr) {
-    // Check if can satisfy sort property with an index
-    auto sort_prop = sort->As<PropertySort>();
-    if (IndexUtil::CheckSortProperty(sort_prop)) {
-      auto indexes = accessor->GetIndexOids(get->GetTableOid());
-      for (auto index : indexes) {
-        if (IndexUtil::SatisfiesSortWithIndex(accessor, sort_prop, get->GetTableOid(), index)) {
-          std::vector<AnnotatedExpression> preds = get->GetPredicates();
-          auto op = std::make_unique<OperatorNode>(
-              IndexScan::Make(db_oid, get->GetTableOid(), index, std::move(preds), is_update,
-                              planner::IndexScanType::AscendingOpenBoth, {})
-                  .RegisterWithTxnContext(context->GetOptimizerContext()->GetTxn()),
-              std::vector<std::unique_ptr<AbstractOptimizerNode>>(), context->GetOptimizerContext()->GetTxn());
-          transformed->emplace_back(std::move(op));
-        }
-      }
-    }
+  // If no sort found in required properties, try optional properties
+  if (sort == nullptr) {
+    sort = context->GetOptionalProperties()->GetPropertyOfType(PropertyType::SORT);
   }
 
-  // Check whether any index can fulfill predicate predicate evaluation
-  if (!get->GetPredicates().empty()) {
-    // Find match index for the predicates
-    auto indexes = accessor->GetIndexOids(get->GetTableOid());
+  bool sort_exists = sort != nullptr;
+  auto sort_prop = sort_exists ? sort->As<PropertySort>() : nullptr;
+  bool sort_possible = sort_exists && IndexUtil::CheckSortProperty(sort_prop);
+  bool predicate_exists = !get->GetPredicates().empty();
+
+  if (predicate_exists) {
+    // Apply predicates if they exist and attempt to use indexes that match at least one predicate
     for (auto &index : indexes) {
       planner::IndexScanType scan_type;
       std::unordered_map<catalog::indexkeycol_oid_t, std::vector<planner::IndexExpression>> bounds;
       std::vector<AnnotatedExpression> preds = get->GetPredicates();
+      // Check whether any index can fulfill in-order predicate evaluation
       if (IndexUtil::SatisfiesPredicateWithIndex(accessor, get->GetTableOid(), get->GetTableAlias(), index, preds,
                                                  allow_cves_, &scan_type, &bounds)) {
-        auto op = std::make_unique<OperatorNode>(IndexScan::Make(db_oid, get->GetTableOid(), index, std::move(preds),
-                                                                 is_update, scan_type, std::move(bounds))
-                                                     .RegisterWithTxnContext(context->GetOptimizerContext()->GetTxn()),
-                                                 std::vector<std::unique_ptr<AbstractOptimizerNode>>(),
-                                                 context->GetOptimizerContext()->GetTxn());
+        // Limit can only be pushed down in the following cases
+        // TODO(dpatra): Limit can be pushed down in the case of an exact scan as well, but the index framework
```

Comment:
I don't really understand this comment. It seems like we are trying to push down limits for exact scans based on the predicate below ...`scan_type == planner::IndexScanType::Exact ||`...

Reply:
I'll move this comment to the index_scan_translator. The function used for exact scans doesn't take advantage of the limit right now.